[acekhoon] iP

acekhoon opened the pull request:
## Cinnamo Chatbot
```
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢀⣀⣀⣀⡀⠀⠀⢀⣀⣀⣀⣀⣀⣄⣀⣀⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣀⣤⡶⠟⠛⠉⠉⣩⠽⠗⠛⠉⠉⠉⠉⠉⠉⠉⠉⠙⠛⠻⠶⢦⣤⣀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢀⣀⣠⡾⠛⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠙⠓⢦⣀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⣀⣴⡮⠗⠋⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠳⣤⢤⣤⣀⠀
⣀⣀⣀⣀⣤⣴⣶⠿⠋⠉⠀⠀⠀⠀⠀⠀⢀⡠⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠢⠈⠙⢷
⠉⠉⠉⠉⠉⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣴⠟⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣼⠋⠀⠀⢀⣴⣿⣶⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢠⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⡇⠀⠀⠀⣾⣿⣿⡿⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢣⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣿⡗⠦⡤⠤⣝⠿⠟⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣀⣀⠀⠀⠀⣰⣿⣿⣆⠀⠀⠀⠀⠈⣧⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢀⣠⠞⠹⣷⣀⣀⣠⠞⠀⠀⠀⠀⠀⠀⠀⢦⣠⣤⣄⠀⢀⣠⠿⣿⣭⣷⣀⠀⣿⣿⣿⡿⠀⠀⠀⠀⠀⢻⠀
⠀⠀⠀⠀⠀⠀⣀⣠⣤⠞⠋⠁⠀⠀⠙⢷⣄⡀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠉⠁⣼⠎⠉⠁⠈⠉⠉⠛⠿⣟⠁⠀⠀⠀⠀⣠⣼⡶
⣤⣤⣴⠶⠾⠛⠋⠁⠀⠀⠀⠀⠀⠀⠀⠀⠉⠛⠷⢦⣄⣀⡀⠀⠀⠀⠀⠀⠀⠀⢠⣟⣽⣶⠀⠀⠀⠀⠀⠀⠀⠈⠳⡶⢶⣴⠟⠉⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢀⣤⡶⠞⠛⠋⠛⠒⠦⡍⠉⣹⠛⠑⠚⠛⠉⠙⠛⠻⢿⣿⣦⣤⣄⣆⠀⠀⠀⠀⠀⣿⣾⣿⠆⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣠⡿⠋⠀⠀⠀⠀⠀⠀⠀⣸⡿⠃⠀⠀⠀⠀⠀⠀⠀⣀⣼⡏⢛⣻⣿⡿⢷⣾⣷⠀⠀⣯⠿⠋⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢰⡿⠀⠀⠀⣷⠀⠀⠀⠀⣼⠏⠀⠀⠀⠀⠀⢒⣶⣶⣾⡟⣙⣷⡼⠿⣿⠀⠀⠈⠻⣤⣾⠻⡀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢹⡇⠀⠀⠀⣿⣤⣶⣤⣼⡏⠀⠀⠀⠀⠀⠀⠺⣷⡏⠘⣿⣿⡿⣷⣤⣿⣧⣀⣀⣴⣏⣀⣀⠙⠦⣄⣀⡀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠘⣷⡀⠀⠀⢸⡏⠁⠃⢹⣷⡄⠀⠀⠀⠀⠀⣰⠿⠀⣠⣿⡿⠃⠹⢿⡞⢻⡏⢩⠟⠋⠉⢹⣧⠀⠀⠉⠉⠙
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠷⣤⣀⢸⣧⠀⠀⠀⠉⠁⠀⠀⠀⠀⢠⡏⠀⣠⣟⡛⠇⠀⠀⣾⠷⣾⠟⠀⠀⠀⠀⢀⣿⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠈⠉⠉⢿⡏⠀⠀⠀⠀⢀⡖⠀⠀⠸⣧⠞⠉⠛⠻⡆⠀⢠⠏⠀⠀⠀⠀⠀⠀⠀⣸⡇⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠒⠒⠒⠚⠻⢦⣄⣀⣴⠾⠿⢶⣶⠶⠶⠶⠶⠶⠤⠿⣶⣿⣶⠶⠶⠿⠷⠶⠶⠾⠯⠤⠀⠀⠀⠀⠀;
```
> The cutest chatbot in the world
## About Cinnamo :laughing: :laughing:
Cinnamo is a chatbot that manages todo tasks for the users so that they could organize their tasks in a more efficient manner. Cinnamo aids the users to improve their productivity by supporting following functions:
1. Can read English text inputs from users
1. Understand commands from users on what type of todo events to note down in the list 
1. Save the todo list in local hard-drive for easy retrieval
1. Friendly and cute messages from Cinnamo for every user inputs! :smiling_face_with_three_hearts:

Here are detailed features provided by Cinnamo:
- [x] Add, delete, list, search, save and update your todo list
- [x] Mark & unmark your tasks
- [x] Indicate the deadline for those tasks

> [!Note]
If you are interested in using Cinnamo to aid your schedule management, you can download the program directly from [here](https://github.com/acekhoon/ip).

## For Aspiring Programmers
If you want to...
- **Add** more useful functionalities to Cinnamo
- **Resolve** potential bugs from the code
- **Make** Cinnamo even more **cute** than current version

then you can participate in this project and improve our Cinnamo chatbot! Simply, you can take source code from [here](https://github.com/acekhoon/ip), improve the bot on your local repo, then ask for the pull request to the main repo! Here's Java code for main method if you are interested! More information of this project is in CS2103T [iP webpage](https://nus-cs2103-ay2324s2.github.io/website/admin/ip-w4.html).

``` ruby
public static void main(String[] args) throws Exception {
        new Duke().run();
    }
```

acekhoon committed:
Level-0 Completed

acekhoon committed:
In progress

acekhoon committed:
Level-2 Updated

acekhoon committed:
Level-3 Updated

acekhoon committed:
level 4 done

acekhoon committed:
Ui Testing Updated

acekhoon committed:
Exceptions are updated

acekhoon committed:
level-6 Done

acekhoon committed:
Enums Updated

acekhoon committed:
File I/O Function Added

acekhoon committed:
duplicate

acekhoon committed:
Level-7 Done

acekhoon committed:
Time Feature Updated

acekhoon committed:
dummy duplicate

acekhoon committed:
Level-8

acekhoon committed:
More OOP Done

acekhoon committed:
Packaging Done

acekhoon committed:
A-Junit Updated Now

acekhoon committed:
JavaDoc comments are updated

acekhoon committed:
dummy

acekhoon committed:
JavaDoc Comments & Headers Updated

acekhoon committed:
Code updated according to coding standard

acekhoon committed:
to-be-merged

acekhoon committed:
empty file deleted

acekhoon committed:
Find function by keyword added

acekhoon committed:
pre-merge commit

acekhoon committed:
Merge with branch-lvl-9

gosongying commented on src/main/java/CinnamoRoll/Deadlines.java:
```diff
@@ -0,0 +1,39 @@
+class Deadlines extends Task {
+
+    private final LocalDateTime deadline;
+    private final DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
```
Is it clearer and more readable if changing format to dateFormat?

gosongying commented on src/main/java/CinnamoRoll/Parser.java:
```diff
@@ -0,0 +1,43 @@
+     * @return A Task object based on the parsed input.
+     * @throws CinnamoTimeException If there is an error parsing the date and time information.
+     */
+    Task parseTasks(String[] arr) throws CinnamoTimeException {
```
Perhaps can use singular form instead of plural form for task because this method is parsing one task only?

gosongying commented on src/main/java/CinnamoRoll/Task.java:
```diff
@@ -0,0 +1,50 @@
+        this.isMarked = false;
+    }
+
+    Task(String str, boolean marked) {
```
Any reason why you don't use isMarked just like the field name for the parameter name 😂, that sounds more like a boolean?

gosongying commented on src/main/java/CinnamoRoll/TaskList.java:
```diff
@@ -0,0 +1,232 @@
+    private final String PATH;
+    private final Parser parser = new Parser();
+
+    private enum Users {
```
Do you think Commands will be more suitable for this variable name?

gosongying commented on src/main/java/CinnamoRoll/TaskList.java:
```diff
@@ -0,0 +1,232 @@
+     *
+     * @throws IOException If there is an error writing to the file.
+     */
+    public void writeInto() throws IOException {
```
Do you think writeIntoFile is clearer for what the method does?

gosongying commented on src/main/java/CinnamoRoll/Task.java:
```diff
@@ -0,0 +1,50 @@
+    /**
+     * Updates the marked status of the tasks stored in the list to be true (marked)
+     */
+    protected void markTask() {
+        this.isMarked = true;
+    }
+
+    /**
+     * Updates the marked status of the tasks stored in the list to be false (unmarked)
+     */
+    protected void unmarkTask() {
+        this.isMarked = false;
+    }
+
+    /**
+     * Retrieve status icon in the form of [(task type)][(marked status)]
+     */
+    abstract String getStatusIcon();
+
+    /**
+     * Prints the description and timing for the task with the status
+     */
+    abstract String addTask(int i);
+
+    /**
+     * Checks whether task's taskname contains the given string for find method
+     */
+    protected boolean containString(String str) {
+        return (this.taskname.contains(str) ? true : false);
+    }
+
+    public String toString() {
+        return this.getStatusIcon() + " " + this.taskname;
+    }
```
Nice naming here, all explain the actions accurately and clearly.

gosongying reviewed (commented):
LGTM. Just a few naming can be clearer but overall not a big deal.

acekhoon committed:
Echo GUi updated

acekhoon committed:
Cinnamo Chatting Function added

acekhoon committed:
Intermediate Step Master for Merge

acekhoon committed:
Level-10 GUi Features Added

acekhoon committed:
Merging back with branch

acekhoon committed:
Improved GUi for Level-10

acekhoon committed:
Checkstyle updated and performed

acekhoon committed:
Asseration feature has been added

Description: assert features for mark, unmark, and add task methods were updated into the chatbot

acekhoon committed:
Codequality improved for some classes

Description: Code quality improvement was done for some classes such as storage, task and tasklist classes

acekhoon committed:
Merge pull request #1 from acekhoon/branch-A-Assertions

Add an asseration feature to verify assumptions on methods behaviours

acekhoon committed:
Merge pull request #2 from acekhoon/branch-A-CodeQuality

Code Quality improved for some classes

acekhoon committed:
C-DetectDuplicates function added

Description: detection of duplicate function is added into this chatbot

acekhoon committed:
Minor bug for reading inputs via loadData fixed

acekhoon committed:
User Guide for the new user added

acekhoon committed:
Minor error was fixed

acekhoon committed:
Minor bugs fixed and finalized

acekhoon committed:
Releases the final chatbot
